packages/prices/test/prices.test.ts

```diff
@@ -0,0 +1,265 @@
+    {
+      price: 25,
+      priceCurrency: "USD",
+      referenceQuantity: {
+        value: 1,
+        unitCode: "DAY",
+      },
+    },
+    {
+      price: 0.25,
+      priceCurrency: "USD",
+      eligibleQuantity: {
+        minValue: 1000,
+        unitCode: "KMT",
+      },
+      referenceQuantity: {
+        value: 1,
+        unitCode: "KMT",
+      },
+    },
+  ];
+  // Case 4: Base price per hour
+  const case4: Offer["priceSpecifications"] = [
+    {
+      price: 5,
+      priceCurrency: "USD",
+      referenceQuantity: {
+        value: 1,
+        unitCode: "HUR",
+      },
+    },
+  ];
+  // Case 5: Different price per day
+  const case5: Offer["priceSpecifications"] = [
+    {
+      price: 25,
+      priceCurrency: "USD",
+      validFrom: "2022-07-01T00:00:00Z",
+      validThrough: "2022-07-10T00:00:00Z",
+      referenceQuantity: {
+        value: 1,
+        unitCode: "DAY",
+      },
+    },
+    {
+      price: 30,
+      priceCurrency: "USD",
+      validFrom: "2022-07-10T00:00:00Z",
+      referenceQuantity: {
+        value: 1,
+        unitCode: "DAY",
+      },
+    },
+  ];
+  // Case 6: Monthly Subscription of X$
+  const case6: Offer["priceSpecifications"] = [
+    {
+      price: 1000,
+      priceCurrency: "USD",
+      billingIncrement: 1,
+      referenceQuantity: {
+        unitCode: "MON",
+      },
+    },
+  ];
+  // Case 7: Discount for Y+ days
+  const case7: Offer["priceSpecifications"] = [
+    {
+      price: 25,
+      priceCurrency: "USD",
+      eligibleQuantity: {
+        maxValue: 3,
+        unitCode: "DAY",
+      },
+      referenceQuantity: {
+        value: 1,
+        unitCode: "DAY",
+      },
+    },
+    {
+      price: 20,
+      priceCurrency: "USD",
+      eligibleQuantity: {
+        minValue: 3,
+        unitCode: "DAY",
+      },
+      referenceQuantity: {
+        value: 1,
+        unitCode: "DAY",
+      },
+    },
+  ];
+
+  const rentalCarReservation: RentalCarReservation = {
+    pickupTime: "2022-07-02T00:00:00Z",
+    dropoffTime: "2022-07-15T00:00:00Z",
+    bookingTime: "",
+    modifiedTime: "",
+    provider: "",
+    underName: "",
+    reservationFor: cid,
+    totalPrice: {
+      price: 20,
+      priceCurrency: "USD",
+    },
+    dropoffLocation: {
+      address: "CA",
+    },
+    pickupLocation: {
+      address: "CA",
+    },
+  };
+
+  const ligoAgreementState: LigoAgreementState = {
+    startOdometer: {
+      value: 1000,
+    },
+    endOdometer: {
+      value: 2000,
+    },
+  };
+
+  //Test Cases
+
+  describe("Case 1: Base Price per day", () => {
+    test("Total Price", async () => {
+      const priceFinder = new Prices();
+      const totalPrice = await priceFinder.CalculateTotalPrice(
+        case1,
+        rentalCarReservation,
+        ligoAgreementState
+      );
+      console.log(totalPrice);
+      expect(totalPrice).toBeDefined();
+      expect(typeof totalPrice === "number").toBe(true);
```

Comment:
Should these each test for a specific total price?

Reply:
It currently only tests if it returns some number. I'll be adding the total sum and new cases soon.